### PR TITLE
feat: Package catalog cli

### PR DESCRIPTION
**Pull request by wtrocki**
PR NR 69 :P 

Adding package catalog idea. as mentioned I'm not sure if those commands actually do not require us to rewrite this CLi. Also do not get actual business value of the current use case. 

This is still WIP but posting for the conversation

### CODE REVIEW

File: src/commands/verify.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { execSync } from "child_process"
+import { readFileSync } from "fs"
+import { existsSync } from "fs"
+import { writeFileSync } from "fs"
+import { resolve } from "path"
+import { PackageFormat } from "../api/PackageFormat"
```
Comment (helio-frota) [inline src/commands/verify.ts:6]: nit changing `api` to `config` fixes the build.

File: src/commands/verify.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import { getPathToPackages } from "../fileHelpers"
```
Comment (helio-frota) [inline src/commands/verify.ts:7]: nit adding `util` to the path fixes the build.

File: src/commands/add.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  const file = readFileSync(pathToFile, "utf8")
+  const packageContent: PackageFormat = JSON.parse(file);
+
+  const response = await prompts([{
+    type: 'text',
+    name: 'category',
+    message: 'Category for your package?'
+  }]);
+
+  packageContent.packages[name] = {
```
Comment (helio-frota) [inline src/commands/add.ts:44]: surrounding with a `if` fixes the build: 

```
if (packageContent.packages) {
    packageContent.packages[name] = {
      ...response
    }
  }
```

**Review by helio-frota — CHANGES_REQUESTED** [review]

File: src/commands/verify.ts (hunk 1 above)
Comment (wtrocki) [reply src/commands/verify.ts:6]:
```suggestion
import { PackageFormat } from "../config/PackageFormat"
```

File: src/commands/verify.ts (hunk 2 above)
Comment (wtrocki) [reply src/commands/verify.ts:7]:
```suggestion
import { getPathToPackages } from "../util/fileHelpers"
```

File: src/commands/add.ts (hunk 3 above)
Comment (wtrocki) [inline src/commands/add.ts:44]:
```suggestion
  packageContent.packages?[name] = {
```

**Comment by alexalikiotis** [comment]
Can someone give me a basic overview of what the package-catalog will do ??

**Comment by wtrocki** [comment]
https://github.com/wtrocki/package-catalog

**Comment by wtrocki** [comment]
@helio-frota Can fix build/tests etc. more interested is that really working for you guys etc.

**Comment by alexalikiotis** [comment]
@wtrocki so is it a kind of `git` for package.json file ??

**Comment by helio-frota** [comment]
@wtrocki sure thing. If that is something that will help with reference architecture like you shared in that [issue](https://github.com/nodeshift/nodejs-reference-architecture/issues/20#issuecomment-706422558). 

I believe yes, because my initial thinking is that we can keep in this project :
- Good to have a more exp person in this project related to TS, since this is our first TS repo under nodeshift organization
- oratrix is already published 
- already has test/ci structure
- Avoid overlap on doing things that are related with package.json
- You mentioned something with licenses and we have a branch that already does the license scanning

**Comment by helio-frota** [comment]
interesting the build is working locally...but still failing on github actions

**Comment by helio-frota** [comment]
oh nice.. that is linting I'm going to created 1 issue for that.

**Comment by helio-frota** [comment]
@wtrocki can you please fix the lint ? you can by pass some `any` for now if needed and we can catch up later :+1:

**Comment by helio-frota** [comment]
my mistake I forgot to add a pretest script to run the lint

**Comment by wtrocki** [comment]
So generally outstanding would be to:

- Improve readme.md
- align style of the cli
- get feedback on validate - detection if package is deprecated can be better.
- fix linting

**Comment by helio-frota** [comment]
@wtrocki when running the cli I found the an error related to `prompts` dependency

```
$ node dist/cli.js 
internal/modules/cjs/loader.js:896
  throw err;
  ^

Error: Cannot find module 'prompts'
```

**Comment by wtrocki** [comment]
Yep.. Sorry. I wanted to create some WIP PR since there is agreement on vision we can fix all issues here and push that forward

**Comment by alexalikiotis** [comment]
I'll look at the PR more closely tomorrow but about the `deprecation` part, you can use the `npm show <package> --json` and this will return a JSON response. If the package is deprecated the JSON will contain a `depreciation` field you can check against (with the value `this version has been deprecated`) instead of relying on regex plus you get extra useful info.

**Comment by wtrocki** [comment]
@alexalikiotis awesome feedback. I have written this package in just 1 hour to showcase idea to the group. Definitely we can work on improving this

**Comment by helio-frota** [comment]
we can ignore the lint for now , PR sent to add a pretest script https://github.com/nodeshift/oratrix/pull/72 :+1:

**Pull request merged**